crates/test_probe/src/main.rs

```diff
@@ -0,0 +1,33 @@
+use composer::api::{Client, Event};
+use eyre::Result;
+use std::{env, thread, time::Duration};
+
+fn main() -> Result<()> {
+    color_eyre::install()?;
+
+    // Get server address, if given
+    let args: Vec<String> = env::args().collect();
+    let arg = args.get(1);
+    let server_address: Option<&str> = arg.map(|x: &String| x.as_str());
+
+    let client = Client::new(server_address)?;
+    let event = Event::TestTick;
+    let delays = {
```

Comment:
nit, optional: `delays_milli_iter`. It's good to include units in scaler variable names. `_iter` is optional, but since you are using a _cool_ trick below, it probably helps to remind the reader of the code what `clone()` is doing there.

Reply:
Or alternatively something like `ms_dalays` or maybe even better encode this in the type system - have this yield `Duration` instead of numeric type.

Reply:
👍 for including units in the variable name. I did that in e9b00bca6b8190abee366611293f94296dacb121 which should address your other remarks as well.